cmd/podman/create.go

```
@@ -391,12 +386,29 @@ func exposedPorts(c *cli.Context, imageExposedPorts map[string]struct{}) (map[na
 
 // Parses CLI options related to container creation into a config which can be
 // parsed into an OCI runtime spec
-func parseCreateOpts(c *cli.Context, runtime *libpod.Runtime, imageName string, data *inspect.ImageData) (*createConfig, error) {
-	var inputCommand, command []string
-	var memoryLimit, memoryReservation, memorySwap, memoryKernel int64
-	var blkioWeight uint16
+func parseCreateOpts(c *cli.Context, runtime *libpod.Runtime, imageName string) (*createConfig, error) {
+	var (
+		inputCommand, command                                    []string
+		memoryLimit, memoryReservation, memorySwap, memoryKernel int64
+		blkioWeight                                              uint16
+	)
+	idmappings, err := ParseIDMapping(c.StringSlice("uidmap"), c.StringSlice("gidmap"), c.String("subuidname"), c.String("subgidname"))
+	if err != nil {
+		return nil, err
+	}
 
+	rtc := runtime.GetConfig()
+	newImage, err := runtime.ImageRuntime().New(imageName, rtc.SignaturePolicyPath, "", os.Stderr, nil, image.SigningOptions{}, false, false)
+	if err != nil {
+		return nil, err
+	}
+	data, err := newImage.Inspect()
```

please return on an error, I've seen a segfault when the error is ignored:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xed29c9]

goroutine 1 [running]:
main.parseCreateOpts(0xc4200129a0, 0xc4200c2370, 0x7ffcc42386d0, 0x6, 0x0, 0x0, 0xc42011f9c0)
	github.com/projectatomic/libpod/cmd/podman/create.go:406 +0x339
main.runCmd(0xc4200129a0, 0x0, 0x0)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xed29c9]

goroutine 1 [running]:
main.parseCreateOpts(0xc4200129a0, 0xc4200c2370, 0x7ffcc42386d0, 0x6, 0x0, 0x0, 0xc42011f9c0)
	github.com/projectatomic/libpod/cmd/podman/create.go:406 +0x339
main.runCmd(0xc4200129a0, 0x0, 0x0)
```